### PR TITLE
Display a table of dhcp snippets

### PR DESCRIPTION
**Pull request by huwshimi**
Done:
- Populated the DHCP table with data from MAAS. Fixes: #74.

QA:
- Vist /dhcp
- Check that you see dhcp data and that you can expand rows, search etc.

### CODE REVIEW

**Comment by lilyvidenova** [comment]
@huwshimi a few comments:

- [ ] When the search field is in focus the clear button seems to be on top of the focus and it looks wrong: 
Screenshot 2019-08-29 at 11 39 53

- [ ] and a few more comments on the screen below: 
Screenshot 2019-08-29 at 11 50 15

**Comment by squidsoup** [comment]
> @huwshimi a few comments:
> 
> * [ ]  When the search field is in focus the clear button seems to be on top of the focus and it looks wrong:
> 
> Screenshot 2019-08-29 at 11 39 53
> 
I pointed that out in a previous branch of @Caleb-Ellis's but apparently that's just the standard vanilla behaviour.

image

You can replicate that here https://docs.vanillaframework.io/patterns/search-box/

Possibly we should file a bug? I can't recall if @Caleb-Ellis did or not.

**Comment by Caleb-Ellis** [comment]
@lilyvidenova @squidsoup yeah this is the bug here: https://github.com/canonical-web-and-design/vanilla-framework/issues/1960

I think we can have a local override that fixes it so long as it links back to this issue.

**Comment by huwshimi** [comment]
@lilyvidenova I've made those changes, would you mind taking another look? Yep that date is in lieu of the username.

File: ui/src/app/settings/views/Dhcp/DhcpList/DhcpList.js [hunk 1]
```diff
@@ -12,61 +17,225 @@ import {
...
+  const deviceItem = devices.find(device => device.system_id === nodeId);
+  const machineItem = machines.find(machine => machine.system_id === nodeId);
+  return {
+    category:
+      (controllerItem && "controller") ||
+      (deviceItem && "device") ||
+      (machineItem && "machine"),
+    node: controllerItem || deviceItem || machineItem
+  };
 };
```
Comment (squidsoup) [inline ui/src/app/settings/views/Dhcp/DhcpList/DhcpList.js]: Just a vague general comment here, it feels like there's a fair bit of complexity in this component, is there any way we can decompose it further? If you don't see any obvious way to manage that, feel free to ignore this.

**Comment by lilyvidenova** [comment]
@huwshimi Just a few comments: 

Screenshot 2019-09-03 at 14 00 53

**Comment by huwshimi** [comment]
@lilyvidenova I've removed that space. Strangely, I bumped the size of that date column, but it's still wrapping for you, maybe there's a font difference or something. I've upped it a bit more.

File: ui/src/app/settings/views/Dhcp/DhcpList/DhcpList.js (hunk 1 above)
Comment (huwshimi) [reply ui/src/app/settings/views/Dhcp/DhcpList/DhcpList.js]: Most of the complexity is for showing the "applies to" links. I think I can move that part into another component which should hugely simplify this one.

**Comment by lilyvidenova** [comment]
Looks great now, thanks @huwshimi 🚀

**Comment by huwshimi** [comment]
Thankyou @lilyvidenova!

**Pull request merged**